rmgpy/data/kinetics/database.py

```diff
@@ -107,86 +107,101 @@ def load(self, path, families=None, libraries=None, depositories=None):
 
     def loadRecommendedFamiliesList(self, filepath):
         """
-        Load the list of recommended families from the given file
-        
-        The file is usually 'kinetics/families/recommended.py'.
-        This is stored as a dictionary of True or False values (checked here),
-        and should contain entries for every available family (checked in loadFamilies).
+        Load the recommended families from the given file.
+        The file is usually stored at 'kinetics/families/recommended.py'.
+
+        The old style was as a dictionary named `recommendedFamilies`
+        containing all family names as keys with True/False values.
+
+        The new style is as multiple sets with unique names which can be
+        used individually or in combination.
+
+        Both styles can be loaded by this method.
         """
-        try:
-            global_context = {}
-            global_context['__builtins__'] = None
-            global_context['True'] = True
-            global_context['False'] = False
-            local_context = {}
-            local_context['__builtins__'] = None
-            f = open(filepath, 'r')
-            exec f in global_context, local_context
-            f.close()
-            self.recommendedFamilies = local_context['recommendedFamilies']
-        except Exception, e:
-            raise DatabaseError('Error while reading list of recommended families from {0}/recommended.py.\n{1}'.format(filepath,e))
-        for recommended in self.recommendedFamilies.values():
-            if not isinstance(recommended, bool):
-                raise DatabaseError("recommendedFamilies dictionary should contain only True or False values")
+        import imp
+
+        # Load the recommended.py file as a module
+        rec = imp.load_source('rec', filepath)
+
+        # For backward compatibility, check for old-style recommendedFamilies dictionary
+        if hasattr(rec, 'recommendedFamilies'):
+            default = set()
+            for family, recommended in rec.recommendedFamilies.iteritems():
+                if recommended:
+                    default.add(family)
+            self.recommendedFamilies = {'default': default}
+        else:
+            recommended = {}
+            for option in [attribute for attribute in dir(rec) if not attribute.startswith('_')]:
```

Comment:
I think this would be slightly more readable if you used `rec.__dict__.iteritems()`.

Reply:
Thanks, I didn't know about `__dict__`. It seems to give the same set of attributes as `dir()` in this case:
```
>>> dir(rec)
['__builtins__', '__doc__', '__file__', '__name__', '__package__', 'default', 'pah']
>>> rec.__dict__.keys()
['__builtins__', '__file__', '__package__', 'default', 'pah', '__name__', '__doc__']
```
I used the list comprehension is to filter out the private/special attributes and only keep the actual variables that have been defined. Using `__dict__` would remove the need for `getattr` though.
```
recommended = {}
for attribute, value in rec.__dict__.iteritems():
    if not attribute.startswith('_'):
        recommended[attribute] = value
```

And with `__dict__`, this could actually be done with a dictionary comprehension:
```
recommended = {attribute: value for attribute, value in rec.__dict__.iteritems() if not attribute.startswith('_')}
```

Let me know what you prefer.

Reply:
I like the dict comprehension